commit 541e3b40b394fb6bde563ff8ce4c882dafca4eb1
Author: Olivier Houchard <ohouchard@haproxy.com>
Date:   Tue Oct 16 18:35:01 2018 +0200

    MINOR: server: Use memcpy() instead of strncpy().
    
    Use memcpy instead of strncpy, strncpy buys us nothing, and gcc is being
    annoying.
    
    (cherry picked from commit 17f8b90736d811ac9a04af198a3aee34e9935cec)
    Signed-off-by: Willy Tarreau <w@1wt.eu>

diff --git a/src/server.c b/src/server.c
index 4941bd03..208f21da 100644
--- a/src/server.c
+++ b/src/server.c
@@ -3078,7 +3078,7 @@ void apply_server_state(void)
 				globalfilepathlen = 0;
 				goto globalfileerror;
 			}
-			strncpy(globalfilepath, global.server_state_base, len);
+			memcpy(globalfilepath, global.server_state_base, len);
 			globalfilepath[globalfilepathlen] = 0;
 
 			/* append a slash if needed */
@@ -3147,7 +3147,7 @@ void apply_server_state(void)
 						localfilepathlen = 0;
 						goto localfileerror;
 					}
-					strncpy(localfilepath, global.server_state_base, len);
+					memcpy(localfilepath, global.server_state_base, len);
 					localfilepath[localfilepathlen] = 0;
 
 					/* append a slash if needed */
